src/context/auth-context.jsx

```diff
@@ -0,0 +1,97 @@
+import { createContext, useReducer, useState } from "react";
+import axios from "axios";
+import { PassWordNotShowIcon } from "../icons/Icons";
+import { AuthReducer } from "../reducer/AuthReducer";
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [userState, setUserState] = useState(false);
+
+  const [state, dispatch] = useReducer(AuthReducer, {
+    field: {},
+    passwordType: "password",
+    showPasswordIcon: <PassWordNotShowIcon />,
+    emailErrState: false,
```

Comment:
why are you using component here? Its better to have logic here

Reply:
i will change this